### PR TITLE
Review service

### PR DESCRIPTION
**Pull request by Mirazyzz**

### CODE REVIEW

File: CheckDrive.Api/CheckDrive.Api/Extensions/DependencyInjection.cs [hunk 1]
```diff
@@ -1,196 +1,135 @@
...
+            .AddPolicy("AdminOrOperator", policy =>
+            {
+                policy.RequireAssertion(context =>
+                    context.User.HasClaim(c => c.Type == "Operator" && c.Value == "true") ||
+                    context.User.HasClaim(c => c.Type == "Admin" && c.Value == "true"));
+            })
+            .AddPolicy("AdminOrDispatcher", policy =>
+            {
+                policy.RequireAssertion(context =>
+                    context.User.HasClaim(c => c.Type == "Dispatcher" && c.Value == "true") ||
```
Comment (FirdavsAX) [inline CheckDrive.Api/CheckDrive.Api/Extensions/DependencyInjection.cs:113]: There we can add name of policies to const strings in new class CheckDrivePolicies or delete there all policies  and use [Authorize(Roles = "Driver,Admin")] attribute.

File: CheckDrive.Api/CheckDrive.Application/Services/Review/DoctorReviewService.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    private async Task<User> GetAndValidateDoctorAsync(Guid doctorId)
+    {
+        var doctor = await _context.Users
+            .FirstOrDefaultAsync(x => x.Id == doctorId && x.Position == EmployeePosition.Doctor);
+
+        if (doctor is null)
+        {
+            throw new EntityNotFoundException($"Doctor with id: {doctorId} is not found.");
+        }
```
[the remark this answers is not in the document]
Comment (aXayitov) [reply CheckDrive.Api/CheckDrive.Application/Services/Review/DoctorReviewService.cs:49]: Are you know KISS principle?
You should write simply, my dear teacher!

File: CheckDrive.Api/CheckDrive.Application/Services/Review/OperatorReviewService.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    private async Task<User> GetAndValidateOperatorAsync(Guid operatorId)
+    {
+        var operatorEntity = await _context.Users
+            .FirstOrDefaultAsync(x => x.Id == operatorId && x.Position == EmployeePosition.Operator);
+
+        if (operatorEntity is null)
+        {
+            throw new EntityNotFoundException($"Operator with id: {operatorId} is not found.");
```
[the remark this answers is not in the document]
Comment (aXayitov) [reply CheckDrive.Api/CheckDrive.Application/Services/Review/OperatorReviewService.cs]: Ohhhh nooooo.

File: CheckDrive.Api/CheckDrive.Application/Services/Review/OperatorReviewService.cs [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            .Where(x => x.Id == checkPointId)
+            .Where(x => x.MechanicHandover != null)
+            .Include(x => x.MechanicHandover)
+            .ThenInclude(mh => mh!.Car)
+            .FirstOrDefaultAsync();
+
+        if (checkPoint is null)
+        {
+            throw new EntityNotFoundException($"Check point with id: {checkPointId} is not found.");
+        }
```
[the remark this answers is not in the document]
Comment (aXayitov) [reply CheckDrive.Api/CheckDrive.Application/Services/Review/OperatorReviewService.cs:56]: -_-

**Pull request merged**